libraries/lib-math/LinearFit.h

```diff
@@ -0,0 +1,41 @@
+#include <cassert>
+#include <numeric>
+#include <utility>
+#include <vector>
+
+template <typename T, typename U>
+std::pair<double, double> LinearFit(
+   const std::vector<T>& x, const std::vector<U>& y, std::vector<double> w = {})
```

Comment:
I wish we had `std::span` from C++20.  I wish we could use `std::vector` less often to pass contiguous ranges of values.  Because `std::vector` specifies storage management details that are extraneous to the algorithm.

Without `span` you would need at least one extra `size_t` argument so that you could pass simple pointers.

Don't fix this but remember it and think about it.